constantdict/__init__.py

```diff
@@ -54,19 +54,16 @@ class constantdict(Dict[K, V]):  # noqa: N801
     def fromkeys(cls: Type[dict[K, V]], *args: Any,
                  **kwargs: Any) -> Any:
         """Create a new :class:`constantdict` from supplied keys and values."""
-        # dict.fromkeys calls __setitem__, hence need to convert
+        # dict.fromkeys calls __setitem__, hence need to convert from a 'dict'
         return cls(dict.fromkeys(*args, **kwargs))
 
     def __hash__(self) -> int:  # type: ignore[override]
         """Return the hash of this :class:`constantdict`."""
         try:
-            return self._hash  # type: ignore[has-type,no-any-return]
+            return self._hash
         except AttributeError:
-            h = 0
-            for key, value in self.items():
-                h ^= hash((key, value))
-            self._hash = h
-            return h
+            self._hash: int = hash(tuple(self.items()))
```

Comment:
@inducer - Is this a correct way to construct the hash? (Interestingly, `hash(tuple(...))` is faster here than `hash(frozenset(...))`)

Reply:
No, this isn't correct, as it may change based on insertion order, but insertion order does not influence dict equality, so this may violate `cdict1 == cdict2 IMPLIES hash(cdict1) == hash(cdict2)`.

Reply:
Good catch, thank you! [7717b29](https://github.com/matthiasdiener/constantdict/pull/17/commits/7717b291b8da21b82b7d14925cb2e5d625de4ed7) (I think) resolves this.